connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -153,12 +156,25 @@ public PluginClassLoader pluginClassLoader(String name) {
         if (inner == null) {
             return null;
         }
-        ClassLoader pluginLoader = inner.get(inner.lastKey());
+        ClassLoader pluginLoader = inner.get(usedPluginDesc(inner));
         return pluginLoader instanceof PluginClassLoader
                ? (PluginClassLoader) pluginLoader
                : null;
     }
 
+    //visible for testing
+    PluginDesc<?> usedPluginDesc(String name) {
+        SortedMap<PluginDesc<?>, ClassLoader> inner = pluginLoaders.get(name);
+        if (inner == null) {
+            return null;
+        }
+        return usedPluginDesc(inner);
```

Comment:
In cases like this, it's probably a bit more idiomatic in Connect to use a tertiary operator so that there is a single `return`:
```suggestion
        return inner == null ? null : usedPluginDesc(inner);
```

Reply:
done